src/plugins/orangehrmAdminPlugin/Api/I18NTranslationAPI.php

```diff
@@ -0,0 +1,170 @@
+
+class I18NTranslationAPI extends Endpoint implements CollectionEndpoint
+{
+    use LocalizationServiceTrait;
+
+    public const PARAMETER_SOURCE_TEXT = 'sourceText';
+    public const PARAMETER_TRANSLATED_TEXT = 'translatedText';
+    public const PARAMETER_GROUP_ID = 'groupId';
+    public const PARAMETER_ONLY_TRANSLATED = 'onlyTranslated';
+
+    public const PARAMETER_RULE_SOURCE_TEXT_MAX_LENGTH = 250;
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $i18NTranslationSearchFilterParams = new I18NTranslationSearchFilterParams();
+        $this->setSortingAndPaginationParams($i18NTranslationSearchFilterParams);
+
+        $i18NTranslationSearchFilterParams->setSourceText(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::PARAMETER_SOURCE_TEXT,
+            )
+        );
+
+        $i18NTranslationSearchFilterParams->setTranslatedText(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::PARAMETER_TRANSLATED_TEXT,
+            )
+        );
+
+        $i18NTranslationSearchFilterParams->setGroupId(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::PARAMETER_GROUP_ID,
+            )
+        );
+
+        $i18NTranslationSearchFilterParams->setOnlyTranslated(
+            $this->getRequestParams()->getBooleanOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::PARAMETER_ONLY_TRANSLATED,
+            )
+        );
+
+        $languageId = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, 'languageId');
+        $i18NTranslationSearchFilterParams->setLanguageId($languageId);
+
+        $translations = $this->getLocalizationService()->getLocalizationDao()->getNormalizedTranslations(
+            $i18NTranslationSearchFilterParams
+        );
+
+        $languageCount = $this->getLocalizationService()->getLocalizationDao()->getTranslationsCount(
+            $i18NTranslationSearchFilterParams
+        );
+
+        return new EndpointCollectionResult(
+            ArrayModel::class,
+            array_values($translations),
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => $languageCount])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                'languageId',
+                new Rule(Rules::POSITIVE)
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_SOURCE_TEXT,
+                    new Rule(Rules::STRING_TYPE),
+                    new Rule(Rules::LENGTH, [!null, self::PARAMETER_RULE_SOURCE_TEXT_MAX_LENGTH])
```

Comment:
seems exclamation put mistakenly here
```suggestion
                    new Rule(Rules::LENGTH, [null, self::PARAMETER_RULE_SOURCE_TEXT_MAX_LENGTH])
```

Reply:
Done